fundingmanager.go

```diff
@@ -2098,8 +2074,18 @@ func (f *fundingManager) sendFundingLocked(peer lnpeer.Peer,
 	// send fundingLocked until we succeed, or the fundingManager is shut
 	// down.
 	for {
-		fndgLog.Debugf("Sending FundingLocked for ChannelID(%v) to "+
-			"peer %x", chanID, peerKey)
+		connected := make(chan lnpeer.Peer, 1)
+		f.cfg.NotifyWhenOnline(peerKey, connected)
```

Comment:
Maybe not that crucial since we can't guarantee that they processed the message?

Reply:
It definitely looks like it should be sync, otherwise the error from that call will be more or less meaningless. The same is true other places we send messages in this file.

It looks like it changed from bing sync to async here: https://github.com/lightningnetwork/lnd/pull/1505

Do you remember why it was changed, or if that was unintentional?

Reply:
yeah the async variant can never return an error, which is fine. most likely the only real errors that would be returned here are shutting down errors or some error that indicates the write failed, e.g. the remote peer disconnects.

it should be fine though to leave it async, as @wpaulino points out, since we can never be sure if they process the message even if we want until it's written

Reply:
The change seems unintentional. Agreed offline that we'll address it as a separate PR.